Framework/API/src/DetectorInfo.cpp

```diff
@@ -32,6 +34,18 @@ DetectorInfo::DetectorInfo(
     m_detIDToIndex[m_detectorIDs[i]] = i;
 }
 
+/// Assigns the contents of the non-wrapping part of `rhs` to this.
```

Comment:
Well, we can assign only information that we own. Things that come from the instrument (or the parameter map) are basically just cached. The point is: `DetectorInfo` does not replace `ParameterMap`, so we cannot replace assignments of the latter with the former.

For example, so we have originally masking, positions, and some efficiency parameter in the parameter map. Oringally assigning the `ParameterMap` took care of all this. Masking and positions can now be accessed via `DetectorInfo`, but in the first step only masking will be stored in the "non-wrapping part" -- positions still forwards to the old instrument.

So if we now assign the `ParameterMap`, it will copy positions and efficiency parameters. To also get masking we assign `DetectorInfo`. If `DetectorInfo` would also do assignments for the "wrapping part" we would get a double assignment for positions.